src/crab/region_semantics.cpp

```
@@ -0,0 +1,77 @@
+                .ub_ceiling = reg.shared_region_size,
+                .ub_message =
+                    std::string("Upper bound must be at most ") + variable_registry.name(reg.shared_region_size)};
+    case T_ALLOC_MEM:
+        return {.lb_floor = LinearExpression{0},
+                .lb_message = "Lower bound must be at least 0",
+                .ub_ceiling = reg.alloc_mem_size,
+                .ub_message = std::string("Upper bound must be at most ") + variable_registry.name(reg.alloc_mem_size)};
+    default:
+        // The caller is expected to gate by region type (e.g., the checker's
+        // ValidAccess switch). Reaching this point is a programming error.
+        assert(false && "region_bounds called on non-region type");
```

Clang (14.0.6)

[warning] redundant boolean literal supplied to boolean operator

(readability-simplify-boolean-expr)